op-chain-ops/upgrades/l1.go

```diff
@@ -173,17 +174,47 @@ func L2OutputOracle(batch *safe.Batch, implementations superchain.Implementation
 		return fmt.Errorf("no initialize method")
 	}
 
-	l2OutputOracleStartingBlockNumber := new(big.Int).SetUint64(config.L2OutputOracleStartingBlockNumber)
-	if config.L2OutputOracleStartingTimestamp < 0 {
-		return fmt.Errorf("L2OutputOracleStartingBlockNumber must be concrete")
+	var l2OutputOracleStartingBlockNumber, l2OutputOracleStartingTimestamp *big.Int
+	var l2OutputOracleProposer, l2OutputOracleChallenger common.Address
+	if config != nil {
+		l2OutputOracleStartingBlockNumber = new(big.Int).SetUint64(config.L2OutputOracleStartingBlockNumber)
+		if config.L2OutputOracleStartingTimestamp < 0 {
+			return fmt.Errorf("L2OutputOracleStartingTimestamp must be concrete")
+		}
+		l2OutputOracleStartingTimestamp = new(big.Int).SetInt64(int64(config.L2OutputOracleStartingTimestamp))
+		l2OutputOracleProposer = config.L2OutputOracleProposer
+		l2OutputOracleChallenger = config.L2OutputOracleChallenger
+	} else {
```

Comment:
What happens if the config is not nil, but a value is potentially invalid?
Maybe prior to executing the upgrade, all deploy configs are checked?

Reply:
Could be enforced through an upgrade lock file or something where only once the deploy config checks are run, the upgrade "lockfile" is written and then `op-upgrade` can be executed atomically?